test_requests.py

```diff
@@ -1079,6 +1079,13 @@ def test_response_iter_lines_reentrant(self):
         next(r.iter_lines())
         assert len(list(r.iter_lines())) == 3
 
+    def test_request_with_float_header(self):
+        """Headers with float value should not result in error."""
+        header = {'wait': 2.5}
+        r = requests.get(httpbin('headers'), headers=header)
+        assert r.status_code == 200
+        assert str(header['wait']) in r.text
```

Comment:
Heh, not quite enough.

httpbin returns JSON. For this web request, it'd return something like this:

```
{
  "args": {}, 
  "headers": {
    "Accept": "*/*", 
    "Accept-Encoding": "gzip, deflate", 
    "Connection": "keep-alive", 
    "Host": "http2bin.org", 
    "User-Agent": "python-requests/2.7.0 CPython/2.7.10 Darwin/14.4.0", 
    "Via": "1.1 http2bin.org", 
    "Wait": "2.5"
  }, 
  "origin": "81.129.190.37", 
  "url": "http://http2bin.org/get"
}
```

So we'd like to assert that the value of the header was right. The assertion is: `assert r.json()['headers']['Wait'] == '2.5'`. (Or something close to it, the capitalisation might be wrong here.